### DISCUSSION TITLE
Css from resources build not in filament default layout

### DISCUSSION THREAD

**Discussion by njxqlus** [filamentphp/filament#7691] [Help] [answered]
My stack:
- laravel 10
- laravel jetstream
- laravel sail
- laravel livewire 3
- filament 3

1. It's fresh install
2. I created `resources/vendor/filament-panels/components/logo.blade.php` like [docs](https://filamentphp.com/docs/3.x/panels/themes#changing-the-logo) said to add logo to filament logo page
3. I added previously not used tailwind class `bg-emerald-400` to the file above
4. I ran `npm run dev`. Logo showed but class not working
5. I ran `npm run build` but result was the same
6. I tried to find this class via chrome devtools but it wasn't in any css
7. I understood that there is no css file from build
8. I found  there there is no `@vite('resources/css/app.css')` in filament base layout from sources

Is it a bug or an expected behaviour?  Am I missing something?

tailwind.config
```js
import forms from '@tailwindcss/forms';
import typography from '@tailwindcss/typography';
import preset from './vendor/filament/support/tailwind.config.preset';


/** @type {import('tailwindcss').Config} */
export default {
    presets: [preset],
    content: [
        './vendor/laravel/framework/src/Illuminate/Pagination/resources/views/*.blade.php',
        './vendor/laravel/jetstream/**/*.blade.php',
        './storage/framework/views/*.php',
        './resources/**/*.blade.php',
        './app/Filament/**/*.php', 
        './vendor/filament/**/*.blade.php',
    ],

    plugins: [forms, typography],
};
```

vite.config
```js
import { defineConfig } from 'vite'
import laravel, { refreshPaths } from 'laravel-vite-plugin'
 
export default defineConfig({
    plugins: [
        laravel({
            input: ['resources/css/app.css', 'resources/js/app.js'],
            refresh: [
                ...refreshPaths,
                'app/Livewire/**',
            ],
        }),
    ],
})
```

**Comment by njxqlus** [comment]
I resolve this problem with registering render hook in **AppServiceProvider**:

```php
use Filament\Support\Facades\FilamentView;
use Illuminate\Support\Facades\Blade;

public function register(): void
    {
        FilamentView::registerRenderHook(
            'panels::head.end',
            fn (): string => Blade::render('@vite(\'resources/css/app.css\')')
        );
        FilamentView::registerRenderHook(
            'panels::scripts.after',
            fn (): string => Blade::render('@vite(\'resources/js/app.js\')')
        );
    }
```

**Reply by danharrin** [reply, marked as answer]
You should use a custom theme instead of doing this

**Reply by njxqlus** [reply]
> Filament allows you to change the CSS used to render the UI by compiling a custom stylesheet to replace the default one. This custom stylesheet is called a "theme".

From what I gathered in the documentation, the theme only adds an extra CSS file and the `@vite('resources/css/app.css')` won't be present in the new theme. What I'm aiming for is to include Tailwind classes from the blade files into the compiled CSS, which I can achieve through the use of the `@vite` directive.

**Reply by njxqlus** [reply]
It looks like you were right. I created a [new theme](https://filamentphp.com/docs/3.x/panels/themes#creating-a-custom-theme), added `'./resources/**/*.blade.php'` to the content section in the new tailwind config and `->viteTheme('resources/css/filament/admin/theme.css')` to the `panel` method in `AdminPanelProvider`. It seems like `viteTheme()` does all the job I need. 

Thanks!
